src/OpenLoco/src/GameCommands/CreateTown.cpp

```diff
@@ -0,0 +1,207 @@
+            for (auto attempts = 200; attempts > 0; attempts--)
+            {
+                uint32_t rand = gameState.rng.randNext();
+                pos = Pos2((((rand >> 16) * kMapColumns) >> 16) << 5, (((rand & 0xFFFF) * kMapRows) >> 16) << 5);
+
+                if (pos.x < 384 || pos.y < 384 || pos.x > 11904 || pos.y > 11904)
+                {
+                    printf("Invalid position (%d, %d) -- trying again\n", pos.x, pos.y);
+                    continue;
+                }
+
+                if (checkSurroundings(pos))
+                {
+                    printf("Surroundings pass the sniff test -- success\n");
+                    foundPos = true;
+                    break;
+                }
+            }
+
+            if (!foundPos)
+            {
+                printf("Unable to find suitable surroundings (-1) -- bailing out\n");
+                return FAILURE;
+            }
+        }
+        else
+        {
+            bool foundPos = false;
+            for (int attempts = 0; attempts < 40; attempts++)
+            {
+                if (attempts == 0)
+                {
+                    uint32_t rand = gameState.rng.randNext();
+                    pos += Pos2(((rand >> 4) & 0xE0) - 0x60, (rand & 0xE0) - 0x60);
+                }
+                else if (attempts <= 10)
+                {
+                    uint32_t rand = gameState.rng.randNext();
+                    pos += Pos2(((rand >> 4) & 0x60) - kTileSize, (rand & 0x60) - kTileSize);
+                }
+
+                if (pos.x < 160 || pos.y < 160 || pos.x > 12128 || pos.y > 12128)
+                {
+                    printf("Invalid position (%d, %d) -- trying again\n", pos.x, pos.y);
+                    continue;
+                }
+
+                if (checkSurroundings(pos))
+                {
+                    printf("Surroundings pass the sniff test -- success\n");
+                    foundPos = true;
+                    break;
+                }
+            }
+
+            if (!foundPos)
+            {
+                printf("Unable to find suitable surroundings (-2) -- bailing out\n");
+                return FAILURE;
+            }
+        }
+
+        newTown = initialiseTown(pos);
+        if (!newTown)
+        {
+            setErrorText(StringIds::too_many_towns);
+            return FAILURE;
+        }
+
+        if (!(flags & Flags::apply))
+        {
+            StringManager::emptyUserString(newTown->name);
+            newTown->name = StringIds::null;
+            return 0;
+        }
+
+        auto currentYear = getCurrentYear();
+        setCurrentYear(currentYear - 51);
+
+        auto growthFactor = args.size * args.size;
+
+        for (int i = 8; i > 0; i--)
+        {
+            for (int j = growthFactor; j > 0; j--)
+            {
+                newTown->grow(0xFF);
+                newTown->recalculateSize();
+            }
+
+            setCurrentYear(currentYear + 7);
+        }
```

Comment:
Not sure about this. It feels like too much growth.

Reply:
I think these kinds of buildings shouldn't be built in-game during 1900 though. Have to check with master.

![Unnamed](https://github.com/OpenLoco/OpenLoco/assets/604665/b28fb765-ae37-40fc-9336-349e93560c07)

Reply:
Found and fixed the mistake.